### DISCUSSION TITLE
Restrict the number of threads for OpenVINO

### DISCUSSION THREAD

**Discussion by cjh1995-ros** [openvinotoolkit/openvino#25118] [Q&A] [answered]
Hi all. Currently I'm developing with OpenVINO for accelerating dl model with intel GPU. 

When I run benchmark_app for my model with following commands, it recorded too many cpu usage.  
`benchmark_app -m model.xml -hint latency -t 20 -d GPU`
![Screenshot from 2024-06-19 17-20-31](https://github.com/openvinotoolkit/openvino/assets/47180415/50b2b97e-7045-4067-8340-3586feb33220)

As you can see

- OPTIMAL_NUMBER_OF_INFER_REQUESTS: 1
- NUM_STREAMS: 1

So I think there's no reason to record 100% CPU usage for it.

Here's how I found the number of threads for benchmark_app.
```bash
$ htop
# almost 100%

$ ps -o nlwp <PID>
# 37
```
![Screenshot from 2024-06-19 17-12-03](https://github.com/openvinotoolkit/openvino/assets/47180415/b7c222d3-d995-4b18-babe-8878dc4d5d08)

Here's my CPU/GPU spec.

``` bash
# CPU
$ lscpu
11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz

# GPU
$ clinfo -l
Platform #0: Intel(R) OpenCL HD Graphics
 -- Device #0: Intel(R) UHD Graphics [0x9a60]
```

Here's my question. 
How can I restrict the number of threads for inference while using GPU? I think it should be small cause I use GPU but it's high. Why so? And Is it possible to restrict the number of threads with 1? 

Thank you

**Comment by geunhwan** [comment, marked as answer]
CPU is mainly consumed by device driver. Could you try GPU_QUEUE_THROTTLE with LOW?

**Reply by cjh1995-ros** [reply]
OH MY GOD.........YOU SAVED MY LIFE! NOW I CAN TUNE PARAMETERS FROM HERE!!!! THANK YOU SO MUCH!
